content/en/docs/advanced-tasks/fail-forward-upgrades.md

```diff
@@ -0,0 +1,123 @@
+---
+title: "Opting into fail forward upgrades"
+linkTitle: "Opting into fail forward upgrades"
+weight: 4
+description: >
+  Fail forward upgrades is a feature that allows for the automatic upgrade of failed installs and upgrades. 
+  It is ultimately unsafe and should only be used in specific circumstances. 
+---
+
+## Warning
+
+Enabling fail forward upgrades may have severe consequences and recovery from these consequences is not guaranteed. Critical versions of operators may fail to install and can thus lead to unknown behavior and potentially data loss. As a result, using this features is not recommended in the majority of circumstances. Only use this feature if you:
+
+- Have complete control over operator installs, including OLM.
+- Do not want to manually upgrade your failed operators.
+
+## What "Fail Forward" upgrades add
+A failed installation/upgrade is typically caused by one of two scenarios:
+
+- **A Failed CSV:** The CSV is in the FAILED phase.
+- **A Failed InstallPlan:** Usually occurs because a resource listed in the InstallPlan fails to be created or updated. An InstallPlan may fail independently of its CSV and may fail to create the CSV.
+
+By opting into "failed forward" upgrades, OLM will allow you to recover from failed installations and upgrades by:
+- Allowing CSVs to move from the FAILED phase to the REPLACING phase when "fail forward" upgrades are enabled.
+- Allowing OLM to calculate new installPlans for a set of installables if:
+  - The InstallPlan referenced by a subscription is in the FAILED phase.
+  - The CatalogSource has been updated to include a new upgrade for one or more CSVs in the namespace.
+
+## Using Fail Forward Upgrades
+
+Since resolution is namespaced, the toggle for allowing "fail forward" upgrades is namespaced as well to avoid having extra burden on the resolver. This removes complexity surrounding the resolver's understanding of which dependent operators can or cannot use the forced upgrade toggle.
```

Comment:
I'm not sure in user-facing documentation we should mention things like "to avoid having extra burden on the resolver" since it's essentially an implementation detail and not something the end user needs to know. It's fair to mention that namespace-scoped resolution is a characteristic of resolution in OLM and therefore all operators in the namespace opt-in to the FailForward strategy if it's set.

Reply:
This is a good callout. I think for now we can remove most of this blurb that mentions the resolver and just let the user know that this is namespace scoped since resolution is namespace scoped.